docs/websites.md

```diff
@@ -179,17 +173,17 @@ resources:
                     Origins:
                         # The website (AWS Lambda)
                         -   Id: Website
-                            DomainName: '#{ApiGatewayRestApi}.execute-api.#{AWS::Region}.amazonaws.com'
-                            # This is the stage, if you are using another one (e.g. prod), you will need to change it here too
-                            OriginPath: '/dev'
+                            DomainName: !Join ['.', [!Ref ApiGatewayRestApi, 'execute-api', !Ref AWS::Region, 'amazonaws.com']]
+                            # This is the stage
+                            OriginPath: "/${opt:stage, 'dev'}"
                             CustomOriginConfig:
                                 OriginProtocolPolicy: 'https-only' # API Gateway only supports HTTPS
                         # The assets (S3)
                         -   Id: Assets
-                            # Use s3-website URLs instead if you host a static website (https://stackoverflow.com/questions/15309113/amazon-cloudfront-doesnt-respect-my-s3-website-buckets-index-html-rules#15528757)
-                            DomainName: '#{Assets}.s3.amazonaws.com'
-                            CustomOriginConfig:
-                                OriginProtocolPolicy: 'http-only' # S3 websites only support HTTP
+                            DomainName: !GetAtt Assets.RegionalDomainName
+                            S3OriginConfig: {} # this key is required to tell CloudFront that this is an S3 origin, even though nothing is configured
+                            # you can also use a CustomOrigin but the S3Origin creates correct 404 error and has other optimizations as well
+                            # https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/DownloadDistS3AndCustomOrigins.html#concept_S3Origin
```

Comment:
Are there any advantages to using `CustomOrigin` instead?

From what you are writing here, it seems there are only downsides. If there are only downsides, then we can remove the last 2 lines of comment. We shouldn't suggest it at all, it's simpler for everyone.

Reply:
I honestly don't know of any advantages of using a CustomOrigin like it was described before.
The only reason (i know of) is that you can use it with an s3 bucket that has website hosting enabled in which case you can configure redirects within the s3 bucket.

I haven't used CustomOrigin for long after reading the docs and switching because it looks like the intended way.
There is also special pricing for traffic from s3 to CloudFront... which is free, i don't know if that works when using a CustomOrigin.

Reply:
But i agree, i remove the comment.
Guess I tried to explain why i changed that within the code instead of within this pull request.
And if you need the other inclusion way, then you probably already know what you are doing.

Reply:
👍  good!

One  last thing I can think of: is it possible to host static websites with `S3OriginConfig`? By that I mean: I want `/` to load `index.html`.

If not, then we should probably keep this bit commented:

```yaml
DomainName: !GetAtt Assets.RegionalDomainName
S3OriginConfig: {} # this key is required to tell CloudFront that this is an S3 origin, even though nothing is configured
# If you host a static website, like a SPA, use s3-website URLs instead of the config above
# See https://stackoverflow.com/questions/15309113/amazon-cloudfront-doesnt-respect-my-s3-website-buckets-index-html-rules#15528757
# DomainName: '<bucket-name>.s3-website-us-east-1.amazonaws.com'
# CustomOriginConfig:
#     OriginProtocolPolicy: 'http-only' # S3 websites only support HTTP
```

Reply:
You can tell CloudFront to deliver a [DefaultRootObject](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-distributionconfig.html#cfn-cloudfront-distribution-distributionconfig-defaultrootobject), although it only works with the actual root of the distribution.

If you want to host a static website and use php just for an api, i'd definitely recommend you use a website origin... although I don't know if it is the best idea to put php behind CloudFront in that case since that's already a compromise to get everything under one domain. An api can live under a separate/generated domain and use the ApiGateway's caching capabilities.

I can still add the documentation about the s3 website hosting though because you can have redirects etc with them which normal s3origins can't.

Although I'd probably document it like this:
```yaml
DomainName: !GetAtt Assets.WebsiteURL
CustomOriginConfig:
    OriginProtocolPolicy: 'http-only' # S3 websites only support HTTP
```

Reply:
:+1: that is perfect! If you can add that in (commented) that's great.

Reply:
Ping me once that is done (I'm not sure I get notified of new commits) and we can merge that.

Reply:
@mnapoli I did a bit of research with s3 website hosting. You'll also need to enable it. I roughly documented it but i don't guarantee it works like that.

Reply:
Yep, that's fine by me, we are giving pointers here, that's not the main use case with Bref.